tools/integration_tests/util/helpers/helpers.go

```diff
@@ -0,0 +1,225 @@
+// If gzipCompress is true, output file is a gzip. The fileSize is the size of the
+// original uncompressed content, and the actual output file size will be
+// that of the compressed file and thus different from fileSize.
+// Caller is responsible for deleting when done.
+// Failure cases:
+// 1. If fileSize <= 0
+// 2. If os.CreateTemp() return error or nil handle
+// 3. If gzip.NewWriter() returned nil handle
+func CreateLocalTempFile(contentSize int, gzipCompress bool) (path string, err error) {
+	if contentSize <= 0 {
+		return "", fmt.Errorf("unsupported fileSize: %d", contentSize)
+	}
+
+	var contentBuilder strings.Builder
+
+	const tempStr = TempFileStrLine + "\n"
+
+	for ; contentSize >= len(tempStr); contentSize -= len(tempStr) {
+		contentBuilder.WriteString(tempStr)
+	}
+
+	if contentSize > 0 {
+		contentBuilder.WriteString(tempStr[0:contentSize])
+	}
+
+	contentSize = contentBuilder.Len()
+	content := contentBuilder.String()
+
+	filenameTemplate := "testfile-*.txt"
+	if gzipCompress {
+		filenameTemplate += ".gz"
+	}
+
+	f, err := os.CreateTemp("/tmp", filenameTemplate)
+	if err != nil {
+		return "", err
+	} else if f == nil {
+		return "", fmt.Errorf("nil file handle returned from os.CreateTemp")
+	}
+
+	filepath := f.Name()
+
+	defer f.Close()
+
+	if gzipCompress {
+		w := gzip.NewWriter(f)
+		if w == nil {
+			return "", fmt.Errorf("failed to open a gzip writer handle")
+		}
+
+		n, err := w.Write([]byte(content))
+		if err != nil {
+			return "", err
+		} else if n != contentSize {
+			return "", fmt.Errorf("failed to write to gzip file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+
+		defer w.Close()
+	} else {
+		n, err := f.WriteString(content)
+		if err != nil {
+			return "", err
+		} else if n != contentSize {
+			return "", fmt.Errorf("failed to write to text file %s. Content-size: %d bytes, wrote = %d bytes", filepath, contentSize, n)
+		}
+	}
+
+	return filepath, nil
+}
+
+// Finds if two files have identical content
+// Needs read permission on files and files
+// to exist.
+// Returns 0 if no error and files match
+// Returns 1 if files don't match
+// Returns 2 if any error
+func DiffFiles(file1, file2 string) (diffval int, err error) {
+	bytes1, err := os.ReadFile(file1)
+	if err != nil || bytes1 == nil {
+		return 2, fmt.Errorf("failed to read file %s", file1)
+	}
+
+	bytes2, err := os.ReadFile(file2)
+	if err != nil || bytes2 == nil {
+		return 2, fmt.Errorf("failed to read file %s", file2)
+	}
+
+	if len(bytes1) != len(bytes2) {
+		return 1, fmt.Errorf("files don't match in size: %s (%d bytes), %s (%d bytes)", file1, len(bytes1), file2, len(bytes2))
+	} else if md5.Sum(bytes1) != md5.Sum(bytes2) {
+		return 1, fmt.Errorf("files don't match in content: %s, %s", file1, file2)
+	}
+
+	return 0, nil
+}
+
+// Executes any given tool (e.g. gsutil/gcloud) with given args.
+func executeToolCommandf(tool string, format string, args ...any) ([]byte, error) {
```

Comment:
The function you pointed to didn't server the purpose (doesn't capture the stderr logs at all, and thus doesn't help debugging in case of a failure in any of the gsutil/gcloud commands; also panics in case of error). So, I added this to generalize all the gsutil/gcloud calls.